protoc-gen-swagger/genswagger/template.go

```diff
@@ -762,13 +780,29 @@ func renderServices(services []*descriptor.Service, paths swaggerPathsObject, re
 					desc := ""
 
 					if len(b.Body.FieldPath) == 0 {
+						// In the proto file, we have body: "*"
 						schema = swaggerSchemaObject{
 							schemaCore: schemaCore{},
 						}
 
 						wknSchemaCore, isWkn := wktSchemas[meth.RequestType.FQMN()]
 						if !isWkn {
-							schema.Ref = fmt.Sprintf("#/definitions/%s", fullyQualifiedNameToSwaggerName(meth.RequestType.FQMN(), reg))
+							var bodyExcludedFields []*descriptor.Field
+							if len(b.PathParams) != 0 {
+								for _, p := range b.PathParams {
+									// We only support excluding top-level fields captured by path parameters.
```

Comment:
We could try to support nested field refs, but it's a fair bit more work for pretty limited return I think.

Reply:
Yeah, lets not do that right now.